contracts/exchange/test/internal.ts

```diff
@@ -439,5 +2357,240 @@ blockchainTests('Exchange core internal functions', env => {
+            expect(logs[1].args.from).to.be.eq(leftOrder.makerAddress);
+            expect(logs[1].args.to).to.be.eq(rightOrder.makerAddress);
+            expect(logs[1].args.amount).bignumber.to.be.eq(matchedFillResults.right.takerAssetFilledAmount);
+
+            // Right maker fee -> right fee recipient
+            expect(logs[2].args.orderHash).to.be.eq(rightOrderHash);
+            expect(logs[2].args.assetData).to.be.eq(rightOrder.makerFeeAssetData);
+            expect(logs[2].args.from).to.be.eq(rightOrder.makerAddress);
+            expect(logs[2].args.to).to.be.eq(rightOrder.feeRecipientAddress);
+            expect(logs[2].args.amount).bignumber.to.be.eq(matchedFillResults.right.makerFeePaid);
+
+            // Left maker fee -> left fee recipient
+            expect(logs[3].args.orderHash).to.be.eq(leftOrderHash);
+            expect(logs[3].args.assetData).to.be.eq(leftOrder.makerFeeAssetData);
+            expect(logs[3].args.from).to.be.eq(leftOrder.makerAddress);
+            expect(logs[3].args.to).to.be.eq(leftOrder.feeRecipientAddress);
+            expect(logs[3].args.amount).bignumber.to.be.eq(matchedFillResults.left.makerFeePaid);
+
+            // Left maker -> taker profit
+            expect(logs[4].args.orderHash).to.be.eq(leftOrderHash);
+            expect(logs[4].args.assetData).to.be.eq(leftOrder.makerAssetData);
+            expect(logs[4].args.from).to.be.eq(leftOrder.makerAddress);
+            expect(logs[4].args.to).to.be.eq(takerAddress);
+            expect(logs[4].args.amount).bignumber.to.be.eq(matchedFillResults.profitInLeftMakerAsset);
+
+            // right maker -> taker profit
+            expect(logs[5].args.orderHash).to.be.eq(rightOrderHash);
+            expect(logs[5].args.assetData).to.be.eq(rightOrder.makerAssetData);
+            expect(logs[5].args.from).to.be.eq(rightOrder.makerAddress);
+            expect(logs[5].args.to).to.be.eq(takerAddress);
+            expect(logs[5].args.amount).bignumber.to.be.eq(matchedFillResults.profitInRightMakerAsset);
+
+            // Right taker fee -> right fee recipient
+            expect(logs[6].args.orderHash).to.be.eq(rightOrderHash);
+            expect(logs[6].args.assetData).to.be.eq(rightOrder.takerFeeAssetData);
+            expect(logs[6].args.from).to.be.eq(takerAddress);
+            expect(logs[6].args.to).to.be.eq(rightOrder.feeRecipientAddress);
+            expect(logs[6].args.amount).bignumber.to.be.eq(matchedFillResults.right.takerFeePaid);
+
+            // Right taker fee -> right fee recipient
+            expect(logs[7].args.orderHash).to.be.eq(leftOrderHash);
+            expect(logs[7].args.assetData).to.be.eq(leftOrder.takerFeeAssetData);
+            expect(logs[7].args.from).to.be.eq(takerAddress);
+            expect(logs[7].args.to).to.be.eq(leftOrder.feeRecipientAddress);
+            expect(logs[7].args.amount).bignumber.to.be.eq(matchedFillResults.left.takerFeePaid);
+        });
+    });
```

Comment:
Would be nice to check the case where `takerFeePaid` can overflow.

Reply:
Same as above